modules/siddhi-core/src/main/java/org/wso2/siddhi/core/executor/function/IfThenElseFunctionExecutor.java

```diff
@@ -81,11 +83,30 @@ public void stop() {
 
     @Override
     public Object[] currentState() {
-        return new Object[0]; //No need to maintain a state.
+        return new Object[0]; // No need to maintain a state.
     }
 
     @Override
     public void restoreState(Object[] state) {
-        //Since there's no need to maintain a state, nothing needs to be done here.
+        // Since there's no need to maintain a state, nothing needs to be done here.
+    }
+
+    @Override
+    public Object execute(ComplexEvent event) {
+        try {
+            Boolean condition = (attributeExpressionExecutors[0].execute(event) != null
+                    && attributeExpressionExecutors[0].execute(event).equals(Boolean.TRUE));
```

Comment:
You can remove first condition and make second condition vise versa so null is handled automatically